packages/mdc-ripple/foundation.js

```diff
@@ -488,17 +485,20 @@ class MDCRippleFoundation extends MDCFoundation {
   /** @private */
   layoutInternal_() {
     this.frame_ = this.adapter_.computeBoundingRect();
-
     const maxDim = Math.max(this.frame_.height, this.frame_.width);
-    const surfaceDiameter = Math.sqrt(Math.pow(this.frame_.width, 2) + Math.pow(this.frame_.height, 2));
 
-    // 60% of the largest dimension of the surface
-    this.initialSize_ = maxDim * MDCRippleFoundation.numbers.INITIAL_ORIGIN_SCALE;
+    // Surface diameter is treated differently for unbounded vs. bounded ripples.
+    // Unbounded ripple diameter is calculated smaller since the surface is expected to already be padded appropriately
+    // to extend the pointer target, and the ripple is expected to meet the edges of the padded hitbox
+    // (which is typically square). Bounded ripples, on the other hand, are fully expected to expand beyond the
+    // surface's longest diameter (calculated based on the diagonal plus a constant padding).
+    this.maxRadius_ = this.adapter_.isUnbounded() ? maxDim :
+      Math.sqrt(Math.pow(this.frame_.width, 2) + Math.pow(this.frame_.height, 2)) + MDCRippleFoundation.numbers.PADDING;
 
-    // Diameter of the surface + 10px
-    this.maxRadius_ = surfaceDiameter + MDCRippleFoundation.numbers.PADDING;
+    // Ripple starts as a fraction of the largest dimension of the surface and scales from 1x to a value based on radius
```

Comment:
Just to clarify, does "1x" refer to the initial starting size (i.e., a fraction of the largest dimension of the surface)?

If so, can we reword this to something like:

```
// Ripple starts as a fraction of the largest dimension of the surface
// and scales up to the calculated maximum radius.
```

Reply:
I was referring to how the CSS initially has scale set to 1 and then scales it up from there to `fgScale_`, which is computed based on the initial size and the max radius. Your rewording doesn't explain why fgScale_ exists, IMO, which is why I added the comment. I'll try re-wording it though...

(TBH I'd love to experiment with actually incorporating .6 and 1 into the CSS scale transform to see if we can simplify the JS even more, but I probably shouldn't invest time on that right now.)